sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/test/ut/blob_checkpoint_store_test.cpp

```diff
@@ -87,23 +117,24 @@ namespace Azure { namespace Messaging { namespace EventHubs { namespace Test {
     EXPECT_EQ(102, checkpoints[0].Offset.Value());
   }
 
-  TEST_F(BlobCheckpointStoreTest, TestOwnerships_LIVEONLY_)
+  TEST_P(BlobCheckpointStoreTest, TestOwnerships)
   {
     std::string const testName = GetRandomName();
-    auto containerClient{Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(
-        GetEnv("CHECKPOINTSTORE_STORAGE_CONNECTION_STRING"), testName, m_blobClientOptions)};
 
-    Azure::Messaging::EventHubs::BlobCheckpointStore checkpointStore(containerClient);
+    auto containerClient{CreateBlobContainerClient(testName)};
 
-    auto ownerships = checkpointStore.ListOwnership(
+    std::shared_ptr<CheckpointStore> checkpointStore{
```

Comment:
Why have you switched to shared_ptr?
If everything is just the same, I propose to keep it as it was, to avoid this very question from the next reviewers, and if it was not, consider putting a comment why, to avoid the very same question :)

Reply:
There are two implementations of this test - one in the eventhubs codebase (which primarily tests the in-memory test checkpoint store) and this one.

By switching to shared_ptr, it allows me to more easily keep the two implementations of the test in sync.